### PR TITLE
feat: add TypeScript/TSX asset handler

### LINKED ISSUE #384: Suport detection of TypeScript
In addition to the support for JavaScript (#362), this would provide explicit support for TypeScript.

Thank you @aabhinavvvvvvv for your interest in implementing this!

### PR DESCRIPTION
**Pull request by aabhinavvvvvvv**
## Summary

Adds explicit TypeScript/TSX asset handler support (resolves #384), following the same pattern as the JavaScript handler (#362).

**Handler detects:**
- **Libraries:** ES module imports (`import x from 'module'`, `import { x } from 'module'`, `import * as x from 'module'`), TypeScript type-only imports (`import type { x } from 'module'`), CommonJS `require()`, and side-effect imports
- **Inputs:** `fs.readFileSync`, `fs.readFile`, `fs.createReadStream`
- **Outputs:** `fs.writeFileSync`, `fs.writeFile`, `fs.appendFile`, `fs.appendFileSync`, `fs.createWriteStream`

**File extensions supported:** `.ts`, `.tsx`, `.mts`, `.cts`

## Changes

- `app/services/assets/handlers/typescript.js` — new handler
- `app/images/ts.svg` — TypeScript icon
- `app/preload.js` — register handler
- `app/utils/workflow.js` — register in dependency resolution and asset type detection
- `app/constants/assets-config.js` — add TS extensions
- `app/components/Workflow/DependencyGraph/CustomNodes/Code/CodeNode.js` — icon mapping
- `app/components/Workflow/DependencyGraph/DependencyGraphEChart.js` — icon mapping
- `test/services/assets/handlers/typescript.spec.js` — 32 tests covering all detection cases
- `test/utils/workflow.spec.js` — add TypeScript assertion to `getAssetType`
- `test/services/assets/asset.spec.js` — add TypeScript file extension tests
- `docs/Assets.md` — document TypeScript support

### CODE REVIEW

File: test/services/assets/handlers/typescript.spec.js [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+        expect(libraries.length).toEqual(1);
+        expect(libraries[0]).toMatchObject({
+          module: 'path',
+        });
+      });
+
+      it('should not detect commented-out imports', () => {
+        const libraries = new TypeScriptHandler().getLibraries(
+          'test.uri',
+          "// import React from 'react';\n// const fs = require('fs');"
```
Comment (lrasmus) [inline test/services/assets/handlers/typescript.spec.js]: These tests all pass, but I thought of a few more test cases to verify - do you mind changing the test to include tests for different whitespace and /* */ comment types?

```
"// import React from 'react';\n// const fs = require('fs');\n//const fs = require('fs');\n  //     const fs = require('fs');\n/*const fs = require('fs');*/"
```

**Review by lrasmus — CHANGES_REQUESTED** [review]
This is really great!  Just a minor request to update some of the test cases (they passed for me, so the code is working great).

File: test/services/assets/handlers/typescript.spec.js (hunk 1 above)
Comment (aabhinavvvvvvv) [reply test/services/assets/handlers/typescript.spec.js]: I've updated the test to cover all those variations

**Review by lrasmus — APPROVED** [review]
Excellent work, thank you!
